### DISCUSSION TITLE
First generation too slow (why?)

### DISCUSSION THREAD

**Discussion by cibernicola** [vladmandic/sdnext#2413] [Q&A] [answered]
I can't say exactly since when, but it's getting worse and worse; the first generation of an image takes too long, too long, several minutes. 
Could it be the Nvidia drivers? Some configuration parameter?

After the first one if it goes better, as always. 

It's a 3090 on Win10 - 48GB RAM - Ryzon 5 3600XT

Any idea?

These numbers do not make any sense:

![image](https://github.com/vladmandic/automatic/assets/4579387/1ebe4bf4-eb0c-48f7-bcb1-f656488afd52)

Thanks!

**Comment by cibernicola** [comment]
https://github.com/vladmandic/automatic/discussions/1285 ?

**Reply by cibernicola** [reply]
Downgrading drivers, now I get Out of memory :) Still searching...

**Reply by MaxJa4** [reply]
How much VRAM is used in the first generation vs the next ones? Is the "3D" percentage at roughly 100% during the first generation? Is the "Copy" percentage at roughly 0% during generation?
Talking about the percentages in the task manager, under the GPU detail view.

**Reply by cibernicola** [reply]
This has -1 sense :S 

```
DDIM Sampler: 100%|████████████████████████████████████████████████████████████████████| 25/25 [00:31<00:00,  1.26s/it]
20:45:54-556761 INFO     Processed: images=1 time=278.84s its=0.09 memory={'ram': {'used': 4.46, 'total': 47.93},
                         'gpu': {'used': 4.94, 'total': 24.0}, 'retries': 0, 'oom': 0}
```

278.84s with 100% GPU 

:(

2nd generation:
```
DDIM Sampler: 100%|████████████████████████████████████████████████████████████████████| 25/25 [00:06<00:00,  3.92it/s]
20:48:42-971053 INFO     Processed: images=1 time=7.00s its=3.57 memory={'ram': {'used': 4.49, 'total': 47.93}, 'gpu':
                         {'used': 14.74, 'total': 24.0}, 'retries': 0, 'oom': 0}
```

And this using controlnet: (now at least works, but);
Running DDIM Sampling with 25 timesteps
```
DDIM Sampler: 100%|████████████████████████████████████████████████████████████████████| 25/25 [00:39<00:00,  1.59s/it]
20:51:09-244936 INFO     Processed: images=1 time=40.41s its=0.62 memory={'ram': {'used': 4.84, 'total': 47.93}, 'gpu':
                         {'used': 23.81, 'total': 24.0}, 'retries': 24, 'oom': 0}
```


Something is wrong and yes, is related with Nvidia drivers :S

**Comment by cibernicola** [comment, marked as answer]
Well, this was "my" solution (extracted from Discord server - forum):

With latest game ready drivers - 545.92, all system back to normal speeds:

![image](https://github.com/vladmandic/automatic/assets/4579387/e4c9d92d-9bb9-4079-96ae-bf762855a07b)
